src/views/ChartPage.vue

```diff
@@ -0,0 +1,463 @@
+      if (masterCategorySelected) {
+        const categories = StoreHandler.getCategoriesByMasterCategory(this.$store, masterCategorySelected, false)
+        for (const category of categories) {
+          listName.push(category.name)
+        }
+      } else {
+        for (const masterCategory of this.$store.state.masterCategories) {
+          const categories = StoreHandler.getCategoriesByMasterCategory(this.$store, masterCategory, false)
+          const archivedCategories = StoreHandler.getCategoriesByMasterCategory(this.$store, masterCategory, true)
+          if (!(categories.length === 0 && archivedCategories.length > 0)) {
+            listName.push(masterCategory.name)
+          }
+        }
+      }
+      return listName
+    },
+    getDatas (type: string, masterCategorySelectedId: string): number[] {
+      const listData: number[] = []
+      const masterCategorySelected = StoreHandler.getMasterCategoryById(this.$store, masterCategorySelectedId)
+      if (masterCategorySelected) {
+        const categories = StoreHandler.getCategoriesByMasterCategory(this.$store, masterCategorySelected, false)
+        for (const category of categories) {
+          switch (type) {
+            case 'allocated':
+              listData.push(Utils.getEurosAmount(this.categoryDataList[category.id]?.allocated))
+              break
+            case 'spent':
+              listData.push(Utils.getEurosAmount(this.categoryDataList[category.id]?.spent * (-1)))
+              break
+            case 'available':
+              listData.push(Utils.getEurosAmount(this.categoryDataList[category.id]?.available))
+              break
+          }
+        }
+      } else {
+        for (const masterCategory of this.$store.state.masterCategories) {
+          let data = 0
+          const categories = StoreHandler.getCategoriesByMasterCategory(this.$store, masterCategory, false)
+          for (const category of categories) {
+            switch (type) {
+              case 'allocated':
+                data += this.categoryDataList[category.id]?.allocated
+                break
+              case 'spent':
+                data += this.categoryDataList[category.id]?.spent * (-1)
+                break
+              case 'available':
+                data += this.categoryDataList[category.id]?.available
+                break
+            }
+          }
+          listData.push(Utils.getEurosAmount(data))
+        }
+      }
+      return listData
+    },
+    getColor (allocated: boolean, spent: boolean, available: boolean, nb: number): string[] {
+      let listColor: string[] = []
+      if (allocated) {
+        listColor.push('#b2babb')
+      }
+      if (spent) {
+        listColor.push('#dc7633')
+      }
+      if (available) {
+        listColor.push('#45c1b8')
+      }
+      if (nb !== 0) {
+        listColor = ['#3498db', '#e74c3c', '#f1c40f']
+        if (nb > 3) {
+          listColor.splice(2, 0, '#27ae60')
```

Comment:
how can you make the purpose of this action more apparant ? naming ? comments ?

Reply:
i removed it